packages/deploy-tooling/src/base/validate.ts

```diff
@@ -387,3 +387,18 @@ async function validateTransportRequestWithAdt(
         }
     }
 }
+
+/**
+ * Returns true if specified destination is on-premise and if environment is App Studio
+ * to show additional info.
+ *
+ * @param destination Indentifier for destination to be checked.
+ * @returns Promise boolean.
+ */
+export async function showAdditionalInfoForOnPrem(destination: string): Promise<boolean> {
+    if (!isAppStudio() || !destination) {
```

Comment:
I would change this to 
``
let showInfo  = false;
if (isAppStudio() && destination) {
    const destinations = await listDestinations();
    showInfo = isOnPremiseDestination(destinations[destination]);
}
return showInfo;
```
You have one exist point to test then.
Also, the code is only executed if it meets the correct conditions which makes it easier to read.